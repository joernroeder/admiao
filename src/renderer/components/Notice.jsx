import styled from '@emotion/styled'
import { background, color } from 'styled-system'

import Text from './Text'

const Notice = styled(Text)`
	${background};
	${color};
`

Notice.defaultProps = {
	p: 1,
	pl: 2,
	mr: -2,
	backgroundColor: 'black',
	color: 'white',
}

export default Notice
