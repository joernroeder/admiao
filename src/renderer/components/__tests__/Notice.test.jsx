import React from 'react'

import { render } from '@testing-library/react'

import Notice from '../Notice'

describe('Notice component', () => {
	it('should render correctly', () => {
		const { container } = render(<Notice>Hello from the other side</Notice>)
		const notice = container.firstChild

		expect(notice).toMatchSnapshot()
	})
})
