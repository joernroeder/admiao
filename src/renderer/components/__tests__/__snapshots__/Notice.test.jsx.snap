// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notice component should render correctly 1`] = `
.emotion-0 {
  background-color: black;
  color: white;
  padding: 4px;
  padding-left: 8px;
  margin-right: -8px;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
  background-color: black;
  color: white;
}

<p
  class="emotion-0"
  color="white"
>
  Hello from the other side
</p>
`;
